File: WeatherGoat/Models/RelativeLocation.cs
using System.Text.Json.Serialization;

namespace WeatherGoat.Models;

public record RelativeLocation
{
    [JsonPropertyName("city")]
    public string City { get; set; }
    
    [JsonPropertyName("state")]
    public string State { get; set; }
    
    [JsonPropertyName("geometry")]
    public string Geometry { get; set; }

    public string CityState => $"{City}, {State}";
}

File: src/models/RelativeLocation.ts
import { JSONProperty, Serializable } from '@depthbomb/serde';

@Serializable()
export class RelativeLocation {
	@JSONProperty()
	public city!: string;

	@JSONProperty()
	public state!: string;

	@JSONProperty()
	public geometry!: string;

	public get cityState() {
		return `${this.city}, ${this.state}`;
	}
}
